viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/tree.tsx

```diff
@@ -74,14 +84,17 @@ export class FilterTree extends React.Component<FilterTreeProps> {
 
     handleCheck = (id: number): void => {
         let checkedIds: number[] = [];
-        const checkedNode = this.getNode(id);
+        const checkedNode = this.getNode(this.props.nodes, id);
         if (checkedNode) {
             if (checkedNode.children.length) {
                 const childrenIds = this.getAllChildrenIds(checkedNode, []);
+                const visibleChildrenIds = childrenIds.filter((id: number) => {
+                    return this.getNode(this.state.filteredNodes, id) !== undefined;
```

Comment:
What about collapsed nodes? Now, if an entry is collapsed, but has children that would be visible with the filter, the children are checked when the parent is checked (for example, in the Disks I/O view, collapse a few devices (like loopX or sdX), filter by "read", and check the trace name at the top, it checks the read entry of the collapsed.

I think I would expect them not to be checked if they are not visible?